compute/cloud-client/src/main/java/compute/ipaddress/AssignStaticExistingVm.java

```diff
@@ -0,0 +1,98 @@
+import com.google.cloud.compute.v1.NetworkInterface;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class AssignStaticExistingVm {
+
+  public static void main(String[] args)
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "your-project-id";
+    // Instance ID of the Cloud project you want to use.
+    String instanceId = "your-instance-id";
+    // name of the zone to create the instance in. For example: "us-west3-b"
+    String zone = "your-zone-id";
+    // Name of the network interface to assign.
+    String netInterfaceName = "your-netInterfaceName-id";
+
+    assignStaticExistingVmAddress(projectId, instanceId, zone, netInterfaceName);
+  }
+
+  // Updates or creates an access configuration for a VM instance to assign a static external IP.
+  // As network interface is immutable - deletion stage is required
+  // in case of any assigned ip (static or ephemeral).
+  // VM and ip address must be created before calling this function.
+  // IMPORTANT: VM and assigned IP must be in the same region.
+  public static Instance assignStaticExistingVmAddress(String projectId, String instanceId,
+                                                       String zone, String netInterfaceName)
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    try (InstancesClient client = InstancesClient.create()) {
+      Instance instance = client.get(projectId, zone, instanceId);
+
+      NetworkInterface networkInterface = null;
+      for (NetworkInterface netInterface : instance.getNetworkInterfacesList()) {
+        if (netInterface.getName().equals(netInterfaceName)) {
+          networkInterface = netInterface;
+        }
+      }
+
+      if (networkInterface == null) {
+        // No network interface named '{network_interface_name}' found on instance {instance_name}.
+        return null;
+      }
+      AccessConfig accessConfig = null;
+      for (AccessConfig config : networkInterface.getAccessConfigsList()) {
+        if (config.getType().equals(AccessConfig.Type.ONE_TO_ONE_NAT.name())) {
+          accessConfig = config;
+        }
+      }
+
+      if (accessConfig != null) {
+        // Delete the existing access configuration first
+        client.deleteAccessConfigAsync(projectId, zone, instanceId,
+                        accessConfig.getName(), netInterfaceName)
+                .get(30, TimeUnit.SECONDS);
+      }
+
+      // Add a new access configuration with the new IP
+      AccessConfig newAccessConfig = AccessConfig.newBuilder()
+              // leave this field undefined to use an IP from a shared ephemeral IP address pool
```

Comment:
Kindly capitalize first letters in the starting word. Applies to multiple places.

```suggestion
              // Leave this field undefined to use an IP from a shared ephemeral IP address pool
```